java/flight/flight-jdbc/src/main/java/org/apache/arrow/jdbc/FlightResultSet.java

```diff
@@ -0,0 +1,1062 @@
+    return ResultSetHelper.getInt(getObject(columnName));
+  }
+
+  @Override
+  public long getLong(String columnName) throws SQLException {
+    return ResultSetHelper.getLong(getObject(columnName));
+  }
+
+  @Override
+  public float getFloat(String columnName) throws SQLException {
+    return ResultSetHelper.getFloat(getObject(columnName));
+  }
+
+  @Override
+  public double getDouble(String columnName) throws SQLException {
+    return ResultSetHelper.getDouble(getObject(columnName));
+  }
+
+  @Deprecated
+  @Override
+  public BigDecimal getBigDecimal(String columnName, int scale) throws SQLException {
+    throw new SQLFeatureNotSupportedException();
+  }
+
+  @Override
+  public byte[] getBytes(String columnName) throws SQLException {
+    return ResultSetHelper.getBytes(getObject(columnName));
+  }
+
+  @Override
+  public Date getDate(String columnName) throws SQLException {
+    return ResultSetHelper.getDate(getObject(columnName));
+  }
+
+  @Override
+  public Time getTime(String columnName) throws SQLException {
+    return ResultSetHelper.getTime(getObject(columnName));
+  }
+
+  @Override
+  public Timestamp getTimestamp(String columnName) throws SQLException {
+    throw new SQLFeatureNotSupportedException();
+  }
+
+  @Override
+  public InputStream getAsciiStream(String columnName) throws SQLException {
+    throw new SQLFeatureNotSupportedException();
+  }
+
+  @Override
+  public InputStream getUnicodeStream(String columnName) throws SQLException {
+    throw new SQLFeatureNotSupportedException();
+  }
+
+  @Override
+  public InputStream getBinaryStream(String columnName) throws SQLException {
+    throw new SQLFeatureNotSupportedException();
+  }
+
+  @Override
+  public SQLWarning getWarnings() throws SQLException {
+    return null;
+  }
+
+  @Override
+  public void clearWarnings() throws SQLException {
+  }
+
+  @Override
+  public String getCursorName() throws SQLException {
+    throw new SQLFeatureNotSupportedException();
+  }
+
+  @Override
+  public ResultSetMetaData getMetaData() throws SQLException {
+    return new FlightResultSetMetaData(this.root.getFieldVectors());
+  }
+
+  @Override
+  public Object getObject(String s) throws SQLException {
+    throw new SQLFeatureNotSupportedException();
+  }
+
+  @Override
+  public int findColumn(String s) throws SQLException {
+    throw new SQLFeatureNotSupportedException();
+  }
+
+  @Override
+  public Reader getCharacterStream(int i) throws SQLException {
+    throw new SQLFeatureNotSupportedException();
+  }
+
+  @Override
+  public Reader getCharacterStream(String s) throws SQLException {
+    throw new SQLFeatureNotSupportedException();
+  }
+
+  @Override
+  public BigDecimal getBigDecimal(int i) throws SQLException {
+    return ResultSetHelper.getBigDecimal(getObject(i));
+  }
+
+  @Override
+  public BigDecimal getBigDecimal(String s) throws SQLException {
+    throw new SQLFeatureNotSupportedException();
```

Comment:
It seems this can also be implemented in a way similar to getBigDecimal(int i)?

Reply:
Yes there are a lot of methods to be implemented still. I just wanted to share an early draft to start a discussion.